lib/collection/src/collection/point_ops.rs

```diff
@@ -154,6 +155,67 @@ impl Collection {
             .unwrap_or_else(|| default_request.with_payload.clone().unwrap());
         let with_vector = request.with_vector;
 
+        let mut order_by = request.order_by.map(OrderBy::from);
+
+        // Handle case of order_by
+        if let Some(order_by) = &mut order_by {
+            // Validate we have a numeric index for the order_by key
+            let has_numeric_index_for_order_by_field = self
+                .payload_index_schema
+                .read()
+                .schema
+                .get(order_by.key.as_str())
+                .is_some_and(|field| field.is_numeric());
+
+            if !has_numeric_index_for_order_by_field {
+                return Err(CollectionError::bad_request(format!(
+                    "No numeric index for order_by key: {}",
+                    order_by.key.as_str()
+                )));
+            }
+
+            // Fetch offset id to get offset value
+            order_by.value_offset = if let Some(offset) = offset {
+                let with_payload_interface =
+                    WithPayloadInterface::Fields(vec![order_by.key.clone()]);
+                let with_payload = WithPayload::from(&with_payload_interface);
+
+                let retrieve_request = Arc::new(PointRequestInternal {
+                    ids: vec![offset],
+                    with_payload: Some(with_payload_interface),
+                    with_vector: false.into(),
+                });
+
+                let offset_point: Vec<_> = {
+                    let shards_holder = self.shards_holder.read().await;
+                    let target_shards = shards_holder.select_shards(shard_selection)?;
+                    let scroll_futures = target_shards.into_iter().map(|(shard, _shard_key)| {
+                        let retrieve_request = retrieve_request.clone();
+                        shard.retrieve(
+                            retrieve_request,
+                            &with_payload,
+                            &with_vector,
+                            read_consistency,
+                            false,
+                        )
+                    });
+
+                    future::try_join_all(scroll_futures).await?
+                };
+
+                // Extract number value from payload
+                offset_point
+                    .iter()
+                    .flatten()
+                    .next()
+                    .and_then(|offset_point| offset_point.payload.as_ref())
+                    .map(|payload| payload.get_value(&order_by.key))
+                    .and_then(|multi_value| multi_value.values().iter().find_map(|v| v.as_f64()))
```

Comment:
Yes, but only if they are in different pages. I considered it to not be expected to work correctly. Otherwise we could only consider first value of multi-value to make paging always work